### PR TITLE
feat: upgrades to `xlsx@0.20.3` via CDN

### PR DESCRIPTION
**Pull request by matthewhartstonge**
Fixes:
- [CVE-2023-30533](https://cdn.sheetjs.com/advisories/CVE-2023-30533)
- [CVE-2024-22363](https://cdn.sheetjs.com/advisories/CVE-2024-22363)

### CODE REVIEW

**Comment by matthewhartstonge** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/46?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#46** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/46?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#43** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/43?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#42** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/42?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#41** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/41?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#40** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/40?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#39** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/39?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/38?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#37** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/37?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#36** <a href="https://app.graphite.dev/github/pr/linc-technologies/ember-spreadsheet-export/36?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @matthewhartstonge and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by matthewhartstonge** [comment]
@hero-david @pauln what are your thoughts on migrating to a git based dep...? 🤢

I'm not really a fan of pulling in a git repo, but it is pinned in the lock file to a specific commit hash. Sheetjs have yeeted themselves into their own world (gitea self hosted git) and removed themselves from pushing the dep up to npm :/

Refer: https://github.com/SheetJS/sheetjs/tree/github?tab=readme-ov-file#thank-you-clippy

Tl;DR: Removes a couple of CVEs.

**Comment by matthewhartstonge** [comment]
The other option is to vendor in the dependency and manually update as CVEs arise..?

**Review by matthewhartstonge — COMMENTED** [review]
Decision over chats was to move forward with using the CDN 👍

**Comment by hero-david** [comment]
> Decision over chats was to move forward with using the CDN 👍

I didn't realise that git was an option 👀 Feels nicer than a tgz

**Comment by matthewhartstonge** [comment]
yarp - give https://github.com/linc-technologies/ember-spreadsheet-export/commit/7b63ce723bb1e931ffdad5e7d9499a71f01ddbb9 a cheeky peep 👀 

Can change back to git on a `v0.16`
